test/unit/nedap/utils/test/api.cljc

```diff
@@ -110,3 +113,70 @@
       (gensym)         42
       (gensym)         nil
       [[1 (gensym) 2]] [[1 (gensym) 3]])))
+
+(deftest expect
+  (let [a (atom 0)]
+    (sut/expect (swap! a inc)
+                :to-change @a
+                :from 0
+                :to 1)
+
+    (sut/expect (swap! a inc)
+                (swap! a inc)
+                :to-change @a
+                :from 1
+                :to 3))
+
+  (testing "the macroexpansion evaluates each part exactly once"
+    (let [proof (atom [])]
+      (sut/expect
+       (swap! proof conj :body)
+       :to-change (do (swap! proof conj :to-change) true)
+       :from      (do (swap! proof conj :from)      true)
+       :to        (do (swap! proof conj :to)        true))
+
+      (is (= [:to-change :from :body :to-change :to]
+             @proof))))
+
+  (testing "exception in body"
+    (is (thrown-with-msg? #?(:clj ExceptionInfo :cljs js/Error) #"my special failure"
```

Comment:
Looks nice, but is it possible to additionally cover that `do-report` does its thing with thrown exceptions as well?

Reply:
The test-reporter would catch this outside of our scope, from science i can confirm it works as expected (:error-report with exception), putting that in a test is non-trivial and i'd rather avoid it :p

Reply:
Currently i walk the argument list in reverse order to construct the hash-maps to maintain both the varargs-bodies and vararg-style options. added extra tests to assert good behaviour

Reply:
Remember that in CLJS you can throw anything. The documentation of [catch](https://cljs.github.io/api/cljs.core/catch) recommends `:default` to catch all kind of errors.